freud/box.pyx

```diff
@@ -300,13 +304,21 @@ cdef class Box:
         vecs = np.asarray(vecs)
         flatten = vecs.ndim == 1
         vecs = np.atleast_2d(vecs)
-        vecs = freud.util._convert_array(vecs, shape=(None, 3)).copy()
+        vecs = freud.util._convert_array(vecs, shape=(None, 3))
+        if out is not None:
+            if out.dtype != np.float32:
+                raise ValueError('expected out array type: np.float32')
```

Comment:
See: https://stackoverflow.com/questions/26921836/correct-way-to-test-for-numpy-dtype
```suggestion
        if out is not None and isinstance(out, np.ndarray) and out.dtype != np.float32:
            raise ValueError('expected out array type: np.float32')
```

Reply:
This has an example of what an invalid out parameter looks like: https://riptutorial.com/numpy/example/10938/the-out-parameter

Here's the NumPy C API that performs error checking: https://github.com/numpy/numpy/blob/cb557b79fa0ce467c881830f8e8e042c484ccfaa/numpy/core/src/multiarray/common.c#L401-L409

I think we're covering our bases reasonably well but just wanted to share that info for reference.

Reply:
I guess if we really want to be thorough we should also check `out.flags['C_CONTIGUOUS']` 🤷

Reply:
I added the check for ```out.flags['C_CONTIGUOUS']```